system-addon/test/unit/utils.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * GlobalOverrider - Utility that allows you to override properties on the global object.
+ *                   Seee unit-entry.js for example usage.
+ */
+class GlobalOverrider {
+  constructor() {
+    this.originalGlobals = new Map();
+    this.addedGlobals = new Set();
+    this.sandbox = sinon.sandbox.create();
+  }
+  _override(key, value) {
+    // Components will throw an error in FF if you override it normal.
+    // So, we have to force override it. Oh well.
+    if (key === "Components") {
+      try {global[key] = value} catch (e) {}
+      return;
+    }
+    if (!this.originalGlobals.has(key)) {
+      this.originalGlobals.set(key, global[key]);
+    }
```

Comment:
It can be overridden more than once, but the `originalGlobals` is what the *original*, unaltered value was. That value should not be re-cached the second time you override it, because it's already been reset once.